Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1

```diff
@@ -189,6 +191,38 @@ function Set-CurrentAzSubscription {
 
     $additional = @{ TenantId = $TenantId }
 
-    Write-Host "##[command] Set-AzContext -SubscriptionId $SubscriptionId $(Format-Splat $additional)"
-    $null = Set-AzContext -SubscriptionId $SubscriptionId @additional
+    Retry-Command -Command 'Set-AzContext' -Verbose -Args `
+    @{
+        SubscriptionId=$SubscriptionId;
+    }
+}
+
+function Retry-Command {
+    param(
+        [Parameter(Mandatory=$true)][string]$command,
+        [Parameter(Mandatory=$true)][hashtable]$args,
+        [Parameter(Mandatory=$false)][int]$retries=5,
+        [Parameter(Mandatory=$false)][int]$secondsDelay=5
+    )
+    
+    $retryCount = 0
+    $completed = $false
+
+    while(-not $completed) {
+        try {
+            Write-Host "##[command]$command $args"
+            & $command @args
+            Write-Verbose("Command [{0}] succeeded." -f $command)
+            $completed = $true
+        } catch {
+            if ($retryCount -ge $retries) {
+                Write-Verbose("Command [{0}] failed the maximum number of {1} times." -f $command, $retryCount)
+                throw
+            } else {
+                Write-Verbose("Command [{0}] failed. Retrying in {1} seconds." -f $command, $secondsDelay)
+                Start-Sleep $secondsDelay
```

Comment:
Can we use an exponential backoff strategy for the delays here? We would like to always use "smart" delays where possible.

Reply:
Yes, exponential backoff can be implemented here - for example:
`$secondsDelay = [math]::Pow(2, retryCount)`. 
Would that be sufficient?

Reply:
Yes, that looks good